_api-reference/cluster-settings.md

```diff
@@ -26,30 +45,64 @@ PUT _cluster/settings
+cluster.routing.allocation.node_initial_primaries_recoveries | Integer | This sets the number of recoveries for unassigned primaries after a node restart. Default is `4`.
+cluster.routing.allocation.same_shard.host | Boolean | When set to `true`, multiple copies of a shard are prevented from being allocated to distinct nodes on the same host. Default is `false`.
+cluster.routing.rebalance.enable | String | Enable or disable rebalancing for specific kinds of shards: <br /> <br /> `all` - Allows shard balancing for all types of shards. <br /> <br /> `primaries` - Allows shard balancing for primary shards only. <br /> <br /> `replicas` - Allows shard balancing for replica shards only. <br /> <br /> `none` - No shard balancing is allowed for any indexes. <br /> <br /> Default is `all`.
+cluster.routing.allocation.allow_rebalance | String | Specify when shard rebalancing is allowed: <br /> <br /> `always` - Always allow rebalancing. <br /> <br /> `indices_primaries_active` - Only allow rebalancing when all primaries in the cluster are allocated. <br /> <br /> `indices_all_active` - Only allow rebalancing when all shards in the cluster are allocated. <br /> <br /> Default is `indices_all_active`.
+cluster.routing.allocation.cluster_concurrent_rebalance | Integer | Allows you to control how many concurrent shard rebalances are allowed across a cluster. Default is `2`.
+cluster.routing.allocation.balance.shard | Floating point | Defines the weight factor for the total number of shards allocated per node. Default is `0.45`.
+cluster.routing.allocation.balance.index | Floating point | Defines the weight factor for the number of shards per index allocated on a node. Default is `0.55`.
+cluster.routing.allocation.balance.threshold | Floating point | The minimal optimization value of operations that should be performed. Default is `1.0`.
+cluster.routing.allocation.disk.threshold_enabled | Boolean | When set to `false`, disables the disk allocation decider. This will also remove any existing `index.blocks.read_only_allow_delete index blocks` when disabled. Default is `true`.
+cluster.routing.allocation.disk.watermark.low | String | This controls the low watermark for disk usage. When set to a percentage, OpenSearch will not allocate shards to nodes that have more than that percentage of disk used. This can also be entered as ratio value, like `0.85`. Finally, this can also be set to a byte value like `400mb`. This setting does not affect the primary shards of newly-created indexes, but will prevent their replicas from being allocated. Default is `85%`.
+cluster.routing.allocation.disk.watermark.high | String | This controls the high watermark. OpenSearch will attempt to relocate shards away from a node whose disk usage is above the percentage defined. This can also be entered as ratio value, like `0.85`. Finally, this can also be set to a byte value like `400mb`. This setting affects the allocation of all shards. Default is `90%`.
```

Comment:
```suggestion
cluster.routing.allocation.disk.watermark.high | String | Controls the high watermark. OpenSearch will attempt to relocate shards away from a node whose disk usage is above the percentage defined. This can also be entered as a ratio value, like `0.85`. Finally, this can also be set to a byte value, like `400mb`. This setting affects the allocation of all shards. Default is `90%`.
```